### ISSUE TITLE
C include format error

### ISSUE THREAD

**Issue by aotto1968** [doxygen/doxygen#11082]
**Describe the bug**
a @include of a **C** file fails with an format error. the problem is a "#" in a string which is wrong parsed

**Screenshots**
![image](https://github.com/user-attachments/assets/cf4a886d-139c-4c76-9480-9de983d17e9b)

**To Reproduce**
use attachment: 
[doxygen_bug__include_c.tar.gz](https://github.com/user-attachments/files/16665038/doxygen_bug__include_c.tar.gz)

**Expected behavior**
no format error

**Version**
> doxygen -V
1.13.0 (b75f8d594a78a986352bda81ce3b94d15d4f4aae)
    with sqlite3 3.42.0.

**Comment by albert-github** [comment]
I think the problem is in the line:
```
  MkBufferCreateLOCAL_T(MkBuffer1024C,buf,0)
```
which should read:
```
  MkBufferCreateLOCAL_T(MkBuffer1024C,buf,0);
```

so a missing `;` and the resulting code is not valid C code.

I think the problem is usage.

**Comment by aotto1968** [comment]
ok, this is your point but it is a macro and the ";" is part of the macro

```
__parser__(constructor,macro,doc-sort=tmpl2)
void MkBufferCreateLOCAL_T(MK_MACRO_TYP cls, MK_MACRO_TYP name, MK_NUM const size);
/// \brief create an new \RMkBufferC object as L)ocal S)torage…
///
/// The \arg{cls} argument is:  \RMkBuffer64S 64 bytes, \RMkBuffer256S = 256 bytes, 
///                             \RMkBuffer1024S = 1024 bytes,
///
/// \par hint
/// access \e string with `bufR.super.buf.storage.first.C`
///
/// \param_MkBufferS_cls
/// \param name   the \e name of the new \RMkBufferC
/// \param size   the initial \e size of the new \RMkBufferC
///
/// \snippet{lineno} kernel_mk.h MkBufferCreateLOCAL_T
// [MkBufferCreateLOCAL_T]
#define MkBufferCreateLOCAL_T(cls,name,size)        \
  cls##R name##R = {0};                             \
  struct MkBufferS * const name = MkBUF(&name##R);  \
  MkBufferInit(name,cls##_TT,size);                 \
// [MkBufferCreateLOCAL_T]
```

**Comment by albert-github** [comment]
The problem here is that doxygen has no idea about the `#define`  (even when the file `kernel_mk.h` is included), doxygen is not a compiler and does not preprocess the code for for showing with `SOURCE_BROWSER=YES` or with the `\include` command.

The problem will also occur when e.g.:
```
MkRtSetup_NULL;
```
is missing the `;`.

The missing of a `;` at the end of a "call" to a defined entry is, to the best of my knowledge, not really advised (and when a define is used in a conditional statement it might give some surprising results) but not against the standard though.

The statements:
```
__parser__(constructor,macro,doc-sort=tmpl2)
void MkBufferCreateLOCAL_T(MK_MACRO_TYP cls, MK_MACRO_TYP name, MK_NUM const size);
```
are also confusing as here it looks like `MkBufferCreateLOCAL_T` is a real function (I assume that `__parser__...` is some directive for a specific compile / preprocessor.

**Comment by aotto1968** [comment]
that is the reason why I near always use the `/usr/bin/source-highlight` tool and use htmlinclude to do syntax-highlighting in doxygen. The core problem is that `doxygen` only support a limit number of languages and even if supported there are a couple of problems.

→ `/usr/bin/source-highlight`  has no problem with the files from above:
→ http://thedev.nhi1.de/theLink/main/perfclient.htm

**Comment by albert-github** [comment]
- Does `source-highlight` exists for al major platforms?
- Does `source-highlight` have the ability to link back to the documentation and vice versa?

**Comment by aotto1968** [comment]
look at → https://www.gnu.org/software/src-highlite/

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #11087

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.13.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
